src/main/cli/cli.c

```diff
@@ -3973,140 +3956,156 @@ STATIC_UNIT_TESTED void cliSet(char *cmdline)
+                        {
+                            // fetch data pointer
+                            int8_t *data = (int8_t *)cliGetValuePointer(val) + i;
+                            // store value
+                            *data = (int8_t)atoi((const char*) valPtr);
+                        }
 
-                            // find next comma (or end of string)
-                            valPtr = strchr(valPtr, ',') + 1;
+                        break;
+                    case VAR_UINT16:
+                        {
+                            // fetch data pointer
+                            uint16_t *data = (uint16_t *)cliGetValuePointer(val) + i;
+                            // store value
+                            *data = (uint16_t)atoi((const char*) valPtr);
+                        }
 
-                            i++;
+                        break;
+                    case VAR_INT16:
+                        {
+                            // fetch data pointer
+                            int16_t *data = (int16_t *)cliGetValuePointer(val) + i;
+                            // store value
+                            *data = (int16_t)atoi((const char*) valPtr);
                         }
-                    }
 
-                    // mark as changed
-                    valueChanged = true;
+                        break;
+                    }
 
-                    break;
+                    // find next comma (or end of string)
+                    valPtr = strchr(valPtr, ',') + 1;
 
+                    i++;
                 }
+            }
 
-                if (valueChanged) {
-                    cliPrintf("%s set to ", val->name);
-                    cliPrintVar(val, 0);
-                } else {
-                    cliPrintErrorLinef("INVALID VALUE");
-                    cliPrintVarRange(val);
-                }
+            // mark as changed
+            valueChanged = true;
 
-                return;
+            break;
+        case MODE_STRING: {
+                char *valPtr = eqptr;
+                valPtr = skipSpace(valPtr);
+
+                const unsigned int len = strlen(valPtr);
+                const uint8_t min = val->config.string.minlength;
+                const uint8_t max = val->config.string.maxlength;
+                const bool updatable = ((val->config.string.flags & STRING_FLAGS_WRITEONCE) == 0 || 
+                                        strlen((char *)cliGetValuePointer(val)) == 0 || 
+                                        strncmp(valPtr, (char *)cliGetValuePointer(val), len) == 0);
+
+                if (updatable && len > 0 && len <= max) {
+                    memset((char *)cliGetValuePointer(val), 0, max);
+                    if (len >= min && strncmp(valPtr, emptyName, len)) {
+                        strncpy((char *)cliGetValuePointer(val), valPtr, len);
+                    }
+                    valueChanged = true;
+                }
```

Comment:
This should not silently ignore the user's actions when they are trying to assign a new value to a write once setting, it should throw an error in this case.

Reply:
It allows the same value to be set (ignored essentially). This was in response to restoring backups, and not having errors.

Reply:
But does it throw an error if the user tries to set a different value?

Reply:
Sure does. Invalid value.